### PR TITLE
fixed analysis issues and added support for dart 3

### PR DESCRIPTION
**Pull request by masnun-siam**
## Description

Fixed the analysis issues and added support for Dart 3

*...*

### CODE REVIEW

File: lib/src/background_usecase.dart [hunk 1]
```diff
@@ -132,8 +132,8 @@ abstract class BackgroundUseCase<T, Params> extends UseCase<T, Params> {
 
   @override
   @nonVirtual
-  Future<Stream<T?>> buildUseCaseStream(params) =>
-      Future.value(Stream.value(null));
+  Stream<T?> buildUseCaseStream(params) =>
```
Comment (ShadyBoukhary) [inline lib/src/background_usecase.dart:135]: Why?

File: example/lib/src/app/pages/home/home_presenter.dart [hunk 2]
```diff
@@ -2,15 +2,15 @@ import '../../../domain/usecases/get_user_usecase.dart';
 import 'package:flutter_clean_architecture/flutter_clean_architecture.dart';
 
 class HomePresenter extends Presenter {
-  Function getUserOnNext;
-  Function getUserOnComplete;
-  Function getUserOnError;
+  Function? getUserOnNext;
```
Comment (ShadyBoukhary) [inline example/lib/src/app/pages/home/home_presenter.dart:5]: Why?

File: example/lib/src/domain/usecases/get_user_usecase.dart [hunk 3]
```diff
@@ -26,7 +27,7 @@ class GetUserUseCase
       // Trigger .onError
       controller.addError(e);
     }
-    return controller.stream;
+    yield* controller.stream;
```
Comment (ShadyBoukhary) [inline example/lib/src/domain/usecases/get_user_usecase.dart:30]: Can you explain the remove of Futures and replacement with this syntax?

**Review by ShadyBoukhary — CHANGES_REQUESTED** [review]
Can you respond to the comments?
The tests are failing due to compilation errors after remove Future<>. You'll need to update the test package.

Also, why did you remove Future<>? I also see you made a lot of variables nullable. Can you explain why?

**Comment by ShadyBoukhary** [comment]
@masnun-siam did you mean to close this? I was going to check it in after the tests pass

**Comment by masnun-siam** [comment]
Sorry about not replying to the comments. Actually, it's been a while since I made the PR and when I saw your comments, I reviewed the changes I made but some of the lines didn't make sense to me either. I actually forgot why I made those changes at the first place. Also, it was one of my first PRs so I didn't check any of the tests before I made the PR so I wanted to close it and recreate it properly on my weekends if the issue still exists.

**Comment by ShadyBoukhary** [comment]
Okay, feel free to create a new one

**Pull request closed without merging**